logutils/getlogger.go

```diff
@@ -0,0 +1,88 @@
+package logutils
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type loggerConfig struct {
+	devMode bool
+	level   string
+}
+
+// LogConfigOption allows to fine-tune the configuration of the logger.
+type LogConfigOption = func(*loggerConfig)
+
+// LogDevMode tells the logger to work in the development mode.
+func LogDevMode(devMode bool) LogConfigOption {
```

Comment:
I'm not a big fan of this generic options pattern, it makes things more complicated (and magic) than they need to be / than is good imo

Reply:
all this magick is to allow very minimal usage like this:

```go
l := logutils.MustGetZapLogger()
defer logutils.FlushZap(l)
```

so that we are not forcing the user to think about log levels, and dev/not-dev (and any other future possible customisable behaviour) before they really want to.